test/e2e/assessment_helpers_test.go

```diff
@@ -351,3 +367,83 @@ func testErrorEmpty(err error) bool {
 		return false
 	}
 }
+
+func assessTestCommands(ctx context.Context, client klient.Client, pod *v1.Pod, testCommands []testCommand) (string, error) {
+	var podlist v1.PodList
+	if err := client.Resources(pod.Namespace).List(ctx, &podlist); err != nil {
+		return "Failed to list pod", err
+	}
+	for _, testCommand := range testCommands {
+		var stdout, stderr bytes.Buffer
+		for _, podItem := range podlist.Items {
+			if podItem.ObjectMeta.Name == pod.Name {
+				//adding sleep time to intialize container and ready for Executing commands
+				time.Sleep(5 * time.Second)
+				if err := client.Resources(pod.Namespace).ExecInPod(ctx, pod.Namespace, pod.Name, testCommand.containerName, testCommand.command, &stdout, &stderr); err != nil {
+					return stderr.String(), err
+				}
+				if !testCommand.testCommandStdoutFn(stdout) {
+					return stderr.String(), fmt.Errorf("Command %v running in container %s produced unexpected output on stdout: %s", testCommand.command, testCommand.containerName, stdout.String())
+				}
+			}
+		}
+	}
+	return "", nil
+}
+
+func provisionPod(ctx context.Context, client klient.Client, t *testing.T, pod *v1.Pod, podState v1.PodPhase, testCommands []testCommand) error {
+	if err := client.Resources().Create(ctx, pod); err != nil {
+		t.Fatal(err)
+	}
+	if err := wait.For(conditions.New(client.Resources()).PodPhaseMatch(pod, podState), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+		t.Fatal(err)
+	}
+	if podState == v1.PodRunning || len(testCommands) > 0 {
+		t.Logf("Waiting for containers in pod: %v are ready", pod.Name)
+		if err := wait.For(conditions.New(client.Resources()).ContainersReady(pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+			//Added logs for debugging nightly tests
+			clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+			if err != nil {
+				t.Fatal(err)
+			}
+			actualPod, err := clientset.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			t.Logf("Expected Pod State: %v", podState)
+			yamlData, err := yaml.Marshal(actualPod.Status)
+			if err != nil {
+				fmt.Println("Error marshaling pod.Status to YAML: ", err.Error())
+			} else {
+				t.Logf("Current Pod State: %v", string(yamlData))
+			}
+			if actualPod.Status.Phase == v1.PodRunning {
```

Comment:
shall `v1.PodRunning` be `podState`? it's same in this case of cause. But looks the algorithm is expecting to use the value passed in.

Reply:
Current code is also OK for me.